Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.16-alpine AS builder
+ARG user=test
+WORKDIR /app
+COPY . .
+ENV GOOS=linux GOARCH=amd64 GOPROXY=https://goproxy.cn
+RUN go build -o reservation_thxx_go \
+    && go build -o reservation_thxx_go_external ./external
+
+FROM ubuntu:20.04 AS runner
+RUN apt-get update && apt-get -y install musl tzdata tar cron mongo-tools
+WORKDIR /app
+COPY --from=builder /app/ .
+EXPOSE 9000
+CMD [ "sh", "-c", "crontab tools/crontab.job && ./reservation_thxx_go --staging=false --web=:9000" ]
```

Comment:
As far as I can remember, the published url linked to port 8080. So I think here should be `--web=:8080` and `EXPOSE 8080` in the above line.

Reply:
The service can be configured to be reverse-proxied by nginx, so an arbitrary port is ok.